common/src/components/Chip/Chip.tsx

```diff
@@ -0,0 +1,67 @@
+import React from "react";
+import { XCircleIcon } from "@heroicons/react/outline";
+import Pill from "../Pill";
+
+export interface ChipProps extends React.HTMLProps<HTMLElement> {
+  /** The style type of the element. */
+  color: "primary" | "secondary" | "neutral";
+  /** The style mode of the element. */
+  mode: "solid" | "outline";
+  /** Handler for clicking the dismiss button in the chip */
+  onDismiss?: React.MouseEventHandler<Element>;
```

Comment:
I looking at @tristan-orourke 's comment [here](https://github.com/GCTC-NTGC/gc-digital-talent/pull/1780#discussion_r787160287) and I'm thinking about this event and what should actually be sent up.  Just the label?  Should there be an `id` prop to have something unique to send back?

Reply:
Actually on a very generic component like this, I'd keep it the way you have it.

Reply:
The parent of this chip component can embed any data it needs in the dismiss handler function itself.
```
someList.map(
  ({id}) => {
    const handler = () => doSomethingWithId(id); // id is now part of the function closure
    return <Chip onDismiss={handler} />;
  }
);
```